pkg/bundle/sync.go

```diff
@@ -141,6 +150,34 @@ func (b *bundle) secretBundle(ctx context.Context, ref *trustapi.SourceObjectKey
 	return string(data), nil
 }
 
+func encodeJKS(trustbundle string) ([]byte, error) {
+	remaining := []byte(trustbundle)
+	ks := jks.New()
```

Comment:
@aidy Sorry to chime in late.  I recently fixed an issue in redhat-cop/cert-utils-operator where the JKS store was updated even when no certs changed. One part of the issue was the missing  `jks.WithOrderedAliases()` option of the jks lib. Once there were many certificates (example: mozilla ca-bundle) in the truststore, the newly generated file was no more binary equivalent to the last one, which lead to updates of ConfigMaps which triggered app restarts due to them listening for changes of the configmaps.



https://github.com/vinzent/cert-utils-operator/blob/cdcb7a30b1bf44a269beeeb016e7ca93af270bd9/controllers/configmaptokeystore/configmap_to_keystore_controller.go#L129-L131

Reply:
Ah. I'd spotted that the output varied, I'd assumed that it was due to including randomness during encoding (as it took a password and you could inject your own crypto).

In any case, we should not have a problem with constant configmap refreshes here, as we don't check for binary equivalence:
https://github.com/cert-manager/trust-manager/pull/122/files#diff-4def846e69c9bdd8e89ae4c9d881e4bfaa70c20b7b024cad43f71268c5f2af77R271-R274

Reply:
@vinzent thank you for that, I didn't know of that option! You're a star!

I think it seems worthwhile adding that for the actual release. More determinism is good. I'll raise a PR for it :grin:

Reply:
for the records, these are the comments where I learned about this:

https://github.com/pavlo-v-chernykh/keystore-go/pull/23#issuecomment-726298997

https://github.com/pavlo-v-chernykh/keystore-go/pull/23#issuecomment-726347215


it works for certificates but not for private-keyse where a random salt is added  every time.